beacon-chain/blockchain/stategenerator/state_generator.go

```diff
@@ -7,37 +7,46 @@ import (
 	"github.com/prysmaticlabs/prysm/beacon-chain/core/state"
 	"github.com/prysmaticlabs/prysm/beacon-chain/db"
 	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
 	"github.com/prysmaticlabs/prysm/shared/hashutil"
+	"github.com/sirupsen/logrus"
 )
 
-// GenerateStateFromSlot generates state from the last finalized epoch till the specified slot.
-func GenerateStateFromSlot(ctx context.Context, db *db.BeaconDB, slot uint64) (*pb.BeaconState, error) {
-	fState, err := db.FinalizedState()
+// GenerateStateFromBlock generates state from the last finalized epoch till the specified block.
```

Comment:
should we update this comment? it's no longer `last finalized epoch` since we are using `db.HistoricalStateFromSlot` correct?

Reply:
ah yes, that should be changed